### DISCUSSION TITLE
Any analog to glDepthRange?

### DISCUSSION THREAD

**Discussion by jwwalker** [google/filament#3991] [Q&A] [answered]
Is there any way in Filament to fool with depth the way OpenGL's `glDepthRange` does?  I mean, short of messing with `gl_FragDepth` in a fragment shader.

**Comment by romainguy** [comment, marked as answer]
There is not, no.

**Comment by jwwalker** [comment]
@romainguy  It turns out that it is possible to emulate `glDepthRange`.  In the material, add a parameter `depthRange` of type `float2`.  And in the vertex shader, say:

```glsl
material.clipSpaceTransform[3][2] = materialParams.depthRange.x +
    materialParams.depthRange.y - 1.0;
material.clipSpaceTransform[2][2] = materialParams.depthRange.y -
    materialParams.depthRange.x;
```

**Reply by romainguy** [reply]
Oh yeah, I meant we don't offer the equivalent of `glDepthRange` on the CPU side.
